packages/form-builder/src/components/canvas/DesignPreview.tsx

```diff
@@ -1,41 +1,53 @@
 import * as React from 'react';
 import {useEffect, useState} from 'react';
-import DesignPreviewLoading from './DesignPreviewLoading';
 
 import Storage from '@givewp/form-builder/common/storage';
 
 import IframeResizer from 'iframe-resizer-react';
 import {useFormState} from '../../stores/form-state';
+import DesignPreviewLoading from "@givewp/form-builder/components/canvas/DesignPreviewLoading";
 
 const DesignPreview = () => {
     const {blocks, settings: formSettings} = useFormState();
-    const [sourceDocument, setSourceDocument] = useState('');
-    const [isLoading, setIsLoading] = useState<boolean>();
+    const [isLoading, setLoading] = useState(true);
+    const [sourceDocument, setSourceDocument] = useState(null);
+    const [previewHTML, setPreviewHTML] = useState(null);
 
     useEffect(() => {
-        setIsLoading(true);
+        setLoading(true);
         Storage.preview({blocks, formSettings}).then((document) => {
             setSourceDocument(document);
-            setIsLoading(false);
         });
     }, [
         JSON.stringify(formSettings),
         JSON.stringify(blocks), // stringify to prevent re-renders caused by object as dep
     ]);
 
-    return isLoading ? (
-        <DesignPreviewLoading />
-    ) : (
-        <IframeResizer
-            srcDoc={sourceDocument}
-            checkOrigin={false} /** The srcDoc property is not a URL and requires that the origin check be disabled. */
-            style={{
-                width: '1px',
-                minWidth: '100%',
-                border: '0',
-            }}
-        />
-    );
+    return (
+        <>
+            {isLoading && <DesignPreviewLoading />}
+            <IframeResizer
+                srcDoc={previewHTML}
+                checkOrigin={false} /** The srcDoc property is not a URL and requires that the origin check be disabled. */
+                style={{
+                    width: '1px',
+                    minWidth: '100%',
+                    border: '0',
+                }}
+            />
+
+            {/* @note This iFrame is used to load and render the design preview document in the background. */}
+            <iframe
+                onLoad={(event) => {
+                    const target = event.target as HTMLIFrameElement;
+                    setPreviewHTML(target.contentWindow.document.documentElement.innerHTML)
+                    setLoading(false)
+                }}
+                srcDoc={sourceDocument}
+                style={{display: 'none'}}
+            />
+        </>
+    )
```

Comment:
This iframe loading part is not necessary since it will be triggered the same time as useEffect.  All you have to do is check if the `sourceDocument` is not null (and revert removal of `setIsLoading(false)` within useEffect:

```
{isLoading && <DesignPreviewLoading />}
{sourceDocument && <IframeResizer
      srcDoc={sourceDocument}
      checkOrigin={false} /** The srcDoc property is not a URL and requires that the origin check be disabled. */
      style={{
          width: '1px',
          minWidth: '100%',
          border: '0',
      }}
  }
  />}
```

Reply:
The `onLoad` even is triggered after the HTML is "ready", which is why I added the event callback. The **rendered** HTML is cached here.